### PR TITLE
feat(registry): metapp in registry support configuration in array format

### PR DESCRIPTION
**Pull request by gene9831**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
  - Enhanced registry data processing to handle array formats for configurations, providing more flexibility in managing registry entries.

- **Tests**
  - Introduced tests to validate the behavior of the new preprocessing feature to ensure reliable handling of various configurations.

### CODE REVIEW

File: packages/register/src/common.js [hunk 1]
```diff
@@ -142,6 +142,30 @@ const handleRegistryProp = (id, value) => {
...
+
+      // normal: { plugins: [ Page, Block, ... ] }
+      // array format: { plugins: [ [ Page, { options: extraOptions } ], Block, ... ] }
+      metaApps.forEach((metaApp, index) => {
+        if (isArrayFormat(metaApp)) {
+          metaApps.splice(index, 1, { ...metaApp[0], ...metaApp[1] })
+        }
+      })
+    })
+}
```
[the remark this answers is not in the document]
Comment (gene9831) [reply packages/register/src/common.js]: @coderabbitai help me to generate the unit testing code of `preprocessRegistry` function

File: packages/register/src/common.js [hunk 2]
```diff
@@ -142,6 +142,30 @@ const handleRegistryProp = (id, value) => {
   }
 }
 
+export const preprocessRegistry = (registry) => {
+  // 元应用支持使用长度为2的数组来配置，第一个参数为元应用，第二个参数是额外的自定义配置。此函数判断数组是否属于这种配置格式
+  const isArrayFormat = (arr) => arr.length === 2 && arr[0].id && !arr[1].id
+
+  Object.entries(registry)
+    .filter(([, metaApps]) => Array.isArray(metaApps))
```
Comment (chilingling) [inline packages/register/src/common.js]: 建议改成 `[_, metaApps]`，这样不容易忽略前面的 `,`

File: packages/register/src/common.js [hunk 3]
```diff
@@ -142,6 +142,30 @@ const handleRegistryProp = (id, value) => {
...
+    .forEach(([type, metaApps]) => {
+      // normal: { layout: Layout }
+      // array format: { layout: [ Layout, { options: extraOptions } ] }
+      if (isArrayFormat(metaApps)) {
+        registry[type] = { ...metaApps[0], ...metaApps[1] }
```
Comment (chilingling) [inline packages/register/src/common.js]: 这里建议限制 `[[app, options]]` 的格式才匹配为配置项。如果是单个的场景，建议手动配置 options 进行覆盖。

Comment (gene9831) [reply packages/register/src/common.js]: 已修改

File: packages/register/src/common.js [hunk 4]
```diff
@@ -142,6 +142,23 @@ const handleRegistryProp = (id, value) => {
...
+    .forEach((metaApps) => {
+      // normal: { plugins: [ Page, Block, ... ] }
+      // array format: { plugins: [ [ Page, { options: extraOptions } ], Block, ... ] }
+      metaApps.forEach((metaApp, index) => {
+        if (isArrayFormat(metaApp)) {
+          metaApps.splice(index, 1, { ...metaApp[0], ...metaApp[1] })
+        }
+      })
+    })
+}
```
[the remark this answers is not in the document]
Comment (gene9831) [reply packages/register/src/common.js]: there is no need to handle nested arrays

File: packages/register/src/common.js [hunk 5]
```diff
@@ -142,6 +142,23 @@ const handleRegistryProp = (id, value) => {
   }
 }
 
+export const preprocessRegistry = (registry) => {
+  // 元应用支持使用长度为2的数组来配置，第一个参数为元应用，第二个参数是额外的自定义配置。此函数判断数组是否属于这种配置格式
+  const isArrayFormat = (arr) => arr.length === 2 && arr[0].id
```
Comment (chilingling) [inline packages/register/src/common.js]: 这里要不要考虑 check 一下是不是数组？ `Array.isArray(arr) && ...`

**Review by chilingling — APPROVED** [review]

**Pull request merged**